src/main/java/com/internet/shop/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.UserDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.model.User;
+import com.internet.shop.service.ShoppingCartService;
+import com.internet.shop.service.UserService;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public User create(User user) {
+        userDao.create(user);
+        ShoppingCart shoppingCart = new ShoppingCart(user.getId());
```

Comment:
should we create the ShoppingCart for all users at the shop?
and for admin too?
the last point at the checklist is
> In main method while testing your code don’t forget to create shoppingCart whenever you create User (Later you will do it during registration of user in controller).

Reply:
fixed